p2p/libp2p/directSender.go

```diff
@@ -161,27 +160,33 @@ func (ds *directSender) Send(topic string, buff []byte, peer p2p.PeerID) error {
 	bufw := bufio.NewWriter(stream)
 	w := ggio.NewDelimitedWriter(bufw)
 
-	go func(msg proto.Message) {
-		err := w.WriteMsg(msg)
-		if err != nil {
-			log.LogIfError(err)
-			_ = stream.Reset()
-			_ = helpers.FullClose(stream)
-			return
-		}
+	err = w.WriteMsg(msg)
+	if err != nil {
+		_ = stream.Reset()
+		_ = helpers.FullClose(stream)
+		return err
+	}
 
-		err = bufw.Flush()
-		if err != nil {
-			log.LogIfError(err)
-			_ = stream.Reset()
-			_ = helpers.FullClose(stream)
-			return
-		}
-	}(msg)
+	err = bufw.Flush()
+	if err != nil {
+		_ = stream.Reset()
+		_ = helpers.FullClose(stream)
+		return err
+	}
 
 	return nil
 }
 
+// SendAsync will asynchronously send a direct message to the connected peer
+func (ds *directSender) SendAsync(topic string, buff []byte, peer p2p.PeerID) {
+	go func() {
+		err := ds.Send(topic, buff, peer)
+		if err != nil {
+			log.Debug(err.Error())
```

Comment:
I would change this to log.Info to see if we still have some errors here

Reply:
changed to log.Info